### PR TITLE
Moved no-receiver size_hint method to a separate trait

### PR DESCRIPTION
**Pull request by jbg**

### CODE REVIEW

**Comment by djc** [comment]
In this one, why isn't `extension()` causing problems? Is it because of the `Sized` bound?

**Comment by jbg** [comment]
Yes. I could move that into the other trait too (and rename the new trait) if desired.

**Comment by jbg** [comment]
(and in that case the Sized bound would no longer be needed)

**Comment by djc** [comment]
So as an alternative, could we add a `Sized` bound for `size_hint()`, as well?

**Comment by jbg** [comment]
It works, but you have to also add a `Self: Sized` bound to `render()`. PR: #271

File: askama/src/lib.rs [hunk 1]
```diff
@@ -502,6 +502,10 @@ pub trait Template {
     fn extension() -> Option<&'static str>
     where
         Self: Sized;
+    fn dyn_size_hint(&self) -> usize;
+}
+
+pub trait TemplateSizeHint {
```
Comment (djc) [inline askama/src/lib.rs]: Let's rename this to `SizedTemplate` and move both `extension()` and `size_hint()` into it?

File: askama/src/lib.rs [hunk 2]
```diff
@@ -502,6 +502,10 @@ pub trait Template {
     fn extension() -> Option<&'static str>
     where
         Self: Sized;
+    fn dyn_size_hint(&self) -> usize;
```
Comment (djc) [inline askama/src/lib.rs]: I would say maybe we can just name this `size_hint()` for now? Not sure if the conflict causes issues in the common case.

Comment (jbg) [reply askama/src/lib.rs]: If they have the same name then importing `SizedTemplate` and trying to use the receiverless `size_hint()` will give a "multiple applicable items in scope" compile error if `Template` is also imported.

Comment (djc) [reply askama/src/lib.rs]: Yeah, I think I'm okay with that.

Comment (jbg) [reply askama/src/lib.rs]: Doesn't it basically make the receiverless `size_hint()` unusable? Or is there a way to disambiguate?

(Giving `extension()` the same treatment causes the same compile error in `testing/tests/simple.rs`.)

Comment (djc) [reply askama/src/lib.rs]: I think you can do something like `SizedTemplate::extension(&t)` or `<t as SizedTemplate>::extension()`.

Comment (jbg) [reply askama/src/lib.rs]: ah yes, it works. I've updated the PR.

**Comment by djc** [comment]
Thanks for sticking with it and working through this with me!

**Comment by djc** [comment]
(Also, I'd be curious to hear a little bit more about what you're using Askama for.)

**Pull request merged**